ambry-commons/src/main/java/com/github/ambry/commons/PipedAsyncWritableChannel.java

```diff
@@ -41,22 +44,29 @@
 public class PipedAsyncWritableChannel implements AsyncWritableChannel {
 
   private final ReadableStreamChannel sourceChannel;
+  private final int secondaryTimeoutInMs;
   private final PipedReadableStreamChannel pipedPrimaryReadChannel;
   private PipedReadableStreamChannel pipedSecondaryReadChannel = null;
   private final ReentrantLock lock = new ReentrantLock();
   private final AtomicBoolean channelOpen = new AtomicBoolean(true);
   private static final Logger logger = LoggerFactory.getLogger(PipedAsyncWritableChannel.class);
+  final static HashedWheelTimer wheel = new HashedWheelTimer(10, TimeUnit.MILLISECONDS, 1024);
```

Comment:
do we want to have a time wheel per channel? Can we create one time wheel for all channels and pass this time wheel to each channel.

Reply:
BTW, let's provide a ThreadFactory to the HashedWheelTimer constructor to rename the worker thread in the wheel.

Reply:
Sure. Provided ThreadFactory to HashedWheelTimer to rename worker threads. Using static to have one time wheel instance for now